src/algorithms/edit_distance.rs

```diff
@@ -1,63 +1,131 @@
-use std::collections::HashMap;
+use crate::data_structures::matrix::Matrix;
 
-#[derive(Eq, PartialEq, Hash)]
-enum Operations {
-    MatchSub,
-    Insert,
-    Delete,
-}
+/// Return the minimum amount of changes needed to transform `s1` into `s2`.
+pub fn edit_distance(s1: &str, s2: &str) -> usize {
```

Comment:
We might care for where changes would need to happen in the future but that has nothing to do with the quality of the code.